.github/workflows/benchmark_trigger.yml

```diff
@@ -0,0 +1,113 @@
+    concurrency:
+      # This cancels queued and in-progress trigger runs for the same PR.
+      group: ${{ github.workflow }}-${{ github.event.number }}
+      cancel-in-progress: true
+    steps:
+      - name: "Checking out repository"
+        # This checkouts from the main branch instead of the pull request. See
+        # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: "Finding the previous workflow run"
+        id: find-workflow
+        env:
+          RUN_JSON: workflow-run.json
+        run: |
+          gh api "/repos/${GITHUB_REPOSITORY}/actions/runs" --method GET -F head_sha="${HEAD_SHA}" \
+            | jq --arg name "${WORKFLOW_NAME}" '.workflow_runs | map(select(.name == $name))[0]' \
+            > "${RUN_JSON}"
+          echo "workflow-url=$(jq --raw-output '.url' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+          echo "run-status=$(jq --raw-output '.status' ${RUN_JSON})" >> "${GITHUB_OUTPUT}"
+      - name: "Cancelling the previous workflow run"
+        # If the workflow isn't completed, we need to cancel it first; otherwise
+        # the API can't rerun it.
+        if: |
+          steps.find-workflow.outputs.workflow-url != 'null' &&
+          steps.find-workflow.outputs.run-status != 'completed'
+        env:
+          IREE_WORKFLOW_RUN_URL: ${{ steps.find-workflow.outputs.workflow-url }}
+        run: build_tools/github_actions/cancel_workflow_and_wait.sh
+      - name: "Getting the latest commit SHA"
+        # A push might have happened to trigger a new workflow run. Check the
+        # PR's latest commit SHA and only rerun if there is no new push.
+        #
+        # It reduces the chance that we rerun a workflow on the previous commit
+        # and cancel (due to the concurrency constraint) the run on the newer
+        # commit. Even if that happens, users will see their CI run fails and
+        # can rerun it manually from the UI.
+        id: get-sha
+        if: steps.find-workflow.outputs.workflow-url != 'null'
+        run: |
+          echo "latest-sha=$(gh api /repos/${GITHUB_REPOSITORY}/pulls/${PR_NUMBER} \
+            | jq --raw-output '.head.sha')" \
+            >> "${GITHUB_OUTPUT}"
+      - name: "Rerequesting the workflow"
+        if: |
+          steps.find-workflow.outputs.workflow-url != 'null' &&
+          steps.get-sha.outputs.latest-sha == env.HEAD_SHA
+        env:
+          WORKFLOW_RUN_URL: ${{ steps.find-workflow.outputs.workflow-url }}
+        run: |
+          RERUN_RESPONSE="$(gh api ${WORKFLOW_RUN_URL}/rerun --method POST \
+            | jq --raw-output '.message' \
+          )" || true
+          if [[ "${RERUN_RESPONSE}" != null \
+                  && "${RERUN_RESPONSE}" != "This workflow is already running" \
+          ]]; then
+            echo "Failed to rerun the workflow: ${RERUN_RESPONSE}"
+            exit 1
```

Comment:
It seems like this condition is backwards?

Reply:
Dropped the logic to ignore the "This workflow is already running" error.

I was thinking about a case where a new run (after our cancellation) starts before we try to rerun the workflow. In this case we "might be" able to ignore the "already running" error. It seems to be too complicated for me now for a rare race condition case